siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+    with open(args.cfg, 'r') as f:
+        config = json.load(f)
+
+    chip = Chip(design='')
+    chip.read_manifest(config["manifest"])
+    new_chip = chip
+    streamlit.session_state['master chip'] = chip
+    streamlit.session_state['job'] = 'default'
+else:
+    chip = streamlit.session_state['master chip']
+    new_chip = Chip(design='')
+    job = streamlit.session_state['job']
+    if job == 'default':
+        new_chip = chip
+    else:
+        new_chip.schema = chip.schema.history(job)
+        new_chip.set('design', chip.design)
+
+with title_col:
+    streamlit.title(f'{new_chip.design} dashboard')
+
+with job_select_col:
+    all_jobs = streamlit.session_state['master chip'].getkeys('history')
+    all_jobs.insert(0, 'default')
+    job = streamlit.selectbox(' ', all_jobs)
+    previous_job = streamlit.session_state['job']
+    streamlit.session_state['job'] = job
+    if previous_job != job:
+        streamlit.session_state['right after rerun'] = True
+        streamlit.experimental_rerun()
+
+# gathering data
+metric_dataframe = report.make_metric_dataframe(new_chip)
+
+# create mapping between task and step, index
+node_to_step_index_map = {}
+for step, index in metric_dataframe.columns.tolist():
+    node_to_step_index_map[step + index] = (step, index)
+
+# concatenate step and index
+metric_dataframe.columns = metric_dataframe.columns.map(lambda x:
+                                                        f'{x[0]}{x[1]}')
+nodes, edges = get_nodes_and_edges(new_chip,
+                                   report.get_flowgraph_edges(new_chip),
+                                   report.get_flowgraph_path(new_chip))
+manifest = report.make_manifest(new_chip)
+
+if 'flowgraph' not in streamlit.session_state:
+    streamlit.session_state['flowgraph'] = True
+
+if os.path.isfile(f'{new_chip._getworkdir()}/{new_chip.design}.png'):
+    tabs = streamlit.tabs(["Metrics",
+                           "Manifest",
+                           "File Preview",
+                           "Design Preview"])
+    metrics_tab, manifest_tab, file_preview_tab, design_preview_tab = tabs
+
+    with design_preview_tab:
+        streamlit.header('Design Preview')
+
+        streamlit.image(f'{new_chip._getworkdir()}/{new_chip.design}.png')
+else:
+    tabs = streamlit.tabs(["Metrics", "Manifest", "File Preview"])
+    metrics_tab, manifest_tab, file_preview_tab = tabs
+
+with metrics_tab:
+    if streamlit.session_state['flowgraph']:
+        node_from_flowgraph, datafram_and_node_info_col = show_flowgraph()
+    else:
+        node_from_flowgraph, datafram_and_node_info_col = dont_show_flowgraph()
+
+    with datafram_and_node_info_col:
+        show_dataframe_header()
+
+        show_dataframe_and_parameter_selection(metric_dataframe)
+
+        streamlit.header('Node Information')
+
+        node_info_cols = streamlit.columns(3, gap='small')
+        metrics_col, records_col, logs_and_reports_col = node_info_cols
```

Comment:
```
metrics_col, records_col, logs_and_reports_col = \
  streamlit.columns(3, gap='small')
```

Reply:
I did this, but this is essentially the same as the magic numbers above. Should I treat this differently?